bin/install-pr

```diff
@@ -9,63 +9,146 @@
+linkerd2_kind_integration_url="https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml"
 
-artifacts=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
-
-archive_url=$(curl -s -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
+# Get the URL for downloading the artifacts archive
+auth="Authorization: token $GITHUB_TOKEN"
+branch=$(curl -sL -H "$auth" "$linkerd2_pulls_url/$pr" | jq -r '.head.ref')
+artifacts=$(curl -sL -H "$auth" "$linkerd2_kind_integration_url/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+archive=$(curl -sL -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
 
+bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
 dir=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
 
 cd "$dir" || exit
 
 echo "### Downloading images ###"
 
-curl -L -o archive.zip -H "$auth" "$archive_url"
-
+curl -L -o archive.zip -H "$auth" "$archive"
 unzip -o archive.zip
 
-echo "### Pushing images ###"
+echo "### Loading images into Docker ###"
 
-for archive in cli-bin cni-plugin controller debug grafana proxy web
-do 
-  image=$(docker load -i "$archive.tar" | sed "s/Loaded image: //")
-  docker push "$image"
-  tag=$(echo "$image" | cut -f 2 -d ":")
-done
+image=$(docker load -i cli-bin.tar | sed 's/Loaded image: //')
+tag=$(echo "$image" | cut -f 2 -d ":")
+
+if [ $is_kind ]
+then
+  # KinD context strings are created by prepending `kind-` to the cluster name
+  # shellcheck disable=SC2086
+  if [ $context ]
+  then
+    name=$(echo "$context" | sed -n -E "s/(kind)-(.*)/\2/p")
+    name_flag="--name $name"
+  fi
+
+  for image in cni-plugin controller debug grafana proxy web
+  do
+    # shellcheck disable=SC2086
+    "$bindir"/kind load image-archive $name_flag $image.tar || touch load_fail &
+  done
+
+  # Wait for `kind load` background processes to complete; exit early if any
+  # job failed
+  wait < <(jobs -p)
+  if [ -f load_fail ]
+  then
+    echo "Loading docker images into KinD cluster failed."
+    exit 1
+  fi
+else
+  for image in cni-plugin controller debug grafana proxy web
+  do
+    docker load -i "$image.tar"
+  done
+fi
 
 cd -
 
 rm -rf "$dir"
 
-linkerd=$(bin/docker-pull-binaries "$tag" | head -n 1)
+case $(uname) in
+  Darwin)
+    platform="darwin"
+    ;;
+  Linux)
+    platform="linux"
+    ;;
+  *)
+    platform="windows"
+    ;;
+esac
+
+linkerd=$("$bindir"/docker-pull-binaries "$tag" | awk -v platform=$platform '$0 ~ platform')
 
 echo "### Pre checks ###"
 
-$linkerd check --pre
+# shellcheck disable=SC2086
+(set -x; "$linkerd" $context_flag check --pre)
 
 echo "### Installing $tag ###"
 
-$linkerd install | kubectl apply -f -
-
-$linkerd check
+# shellcheck disable=SC2086
+(set -x; "$linkerd" $context_flag install | kubectl $context_flag apply -f -)
+# shellcheck disable=SC2086
+(set -x; "$linkerd" $context_flag check)
```

Comment:
You can actually just do `set -x` one more time like:
```bash
(
  set -x
  "$linkerd" $context_flag check --pre
   # shellcheck disable=SC2086
  "$linkerd" $context_flag install | kubectl $context_flag apply -f -
   # shellcheck disable=SC2086
  "$linkerd" $context_flag check
)
```

Reply:
I'd like to keep the `echo "### Installing $tag ###"`, but I can wrap the commands after in parens.